engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -101,13 +104,13 @@ public T getValue() {
     }
 
     public boolean setValue(T newValue) {
-        if (!validator.validate(newValue))
+        if (!validator.validate(newValue)) {
+            LOGGER.warn("The passed value {} is invalid.", newValue);
```

Comment:
Ah, that's what you meant. =)

No, this isn't good: it's too generic, it wouldn't help you debug a failure. The validator itself has to log a warning and then return false, so that the warning can be very specific on what went wrong, i.e. the new value being beyond the maximum value allowed.